File: top_100_liked_algos_and_data_structures/find_missing_number_in_array/missing_element.c
#include <stdio.h>

int missing_element(int list[], int n) {

    int total = 0;
    for (int i = 0; i < n; i++) {total += list[i];}
    return (n + 1) + n *(n + 1) / 2 - total;

}

int main() {
    int list[] = {3, 4, 2, 5, 6};
    int n = sizeof(list)/sizeof(list[0]);
    printf("The missing element is %d\n",missing_element(list, n) );
}

File: top_100_liked_algos_and_data_structures/find_missing_number_in_array/missing_element.rs
fn missing_element(list: &[i32]) -> i32 {
    let n = list.len() as i32;
    let total: i32 = list.iter().sum();

    return (n + 1) + n *(n + 1) / 2 - total;
}

fn main() {
    let arr: [i32;5] = [3, 2, 4, 6, 1];
    println!("The missing element is {}", missing_element(&arr))
}